StackExchange.Redis/StackExchange/Redis/ResultProcessor.cs

```diff
@@ -1301,6 +1325,461 @@ protected override bool SetResultCore(PhysicalConnection connection, Message mes
+                                                            idleTimeInMs,
+                                                            deliveryCount);
+                }
+
+                SetResult(message, messages);
+                return true;
+            }
+        }
+
+        internal abstract class StreamProcessorBase<T> : ResultProcessor<T>
+        {
+            // For command response formats see https://redis.io/topics/streams-intro.
+
+            protected RedisStreamEntry[] ParseRedisStreamEntries(RawResult result)
+            {
+                if (result.Type != ResultType.MultiBulk)
+                {
+                    return null;
+                }
+
+                var arr = result.GetItems();
+
+                var entries = new RedisStreamEntry[arr.Length];
+
+                for (var i = 0; i < arr.Length; i++)
+                {
+                    if (arr[i].IsNull || arr[i].Type != ResultType.MultiBulk)
+                    {
+                        entries[i] = RedisStreamEntry.Null;
+                    }
+                    else
+                    {
+                        // Process the Multibulk array for each entry. The entry contains the following elements:
+                        //  [0] = SimpleString (the ID of the stream entry)
+                        //  [1] = Multibulk array of the name/value pairs of the stream entry's data
+                        var item = arr[i].GetItems();
+
+                        var id = item[0].AsRedisValue();
+                        var values = ParseStreamEntryValues(item[1]);
+
+                        entries[i] = new RedisStreamEntry(id, values);
+                    }
+                }
+                return entries;
+            }
+
+            protected NameValueEntry[] ParseStreamEntryValues(RawResult result)
+            {
+                // The XRANGE, XREVRANGE, XREAD commands return stream entries
+                // in the following format.  The name/value pairs are interleaved
+                // in the same fashion as the HGETALL response.
+                // 
+                // 1) 1) 1518951480106-0
+                //    2) 1) "sensor-id"
+                //       2) "1234"
+                //       3) "temperature"
+                //       4) "19.8"
+                // 2) 1) 1518951482479-0
+                //    2) 1) "sensor-id"
+                //       2) "9999"
+                //       3) "temperature"
+                //       4) "18.2"
+
+
+                if (result.Type != ResultType.MultiBulk)
+                {
+                    return null;
+                }
+
+                var arr = result.GetItems();
+
+                if (arr == null)
+                {
+                    return null;
+                }
+
+                NameValueEntry[] pairs;
+
+                int count = arr.Length / 2;
+
+                if (count == 0)
+                {
+                    pairs = new NameValueEntry[0];
```

Comment:
`Array.Empty<NameValueEntry>()`

Also: would just `return` directly here and in the else, keep it simple!

Note: it's okay to drop `net45` from the `.csproj` to get this to compile...we're already on `netstandard2.0`-only in the `pipelines` branch already, so no loss :)